### ISSUE TITLE
[a11y] Detected Screen Reader Issues

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#16411]
Some areas of Dashboard have been reported to be lacking narration by a screen reader.

### Side Nav

Issue: Now when menu option getting expanded/collapsed then same is not getting announced by screen reader.

### Tabs

Issue: On pressing up/down arrow keys links are getting selected but same is not getting announced by screen reader."

### Radio Groups

Radio button group text still not getting announced by screen reader.

**Comment by yonasberhe23** [comment]
Requires manual testing. Not an automation candidate - screen reader narration cannot be automated

**Comment by a-arias** [comment]
**Tests**
Fix validated ✅
Rancher : v2.14-03c65adbbe9dedb99b1a938799b506ff390b3cf6-head
UI: master 20127be

OS: MacOs 26.2 (25C56)
Browsers: Chrome 144

OS: Windows 11
Browsers: Chrome 144

We have some delay using narrator in windows in some scenarios.

*Screenshots**

**Cluster: Create, Edit configuration tabs behavior meets the expectation.**

https://github.com/user-attachments/assets/a24f3fd5-da43-425d-81e7-dba36e66a76f

**Cluster: Create, Edit specific to node pools behavior regressions weren't observed.**

https://github.com/user-attachments/assets/7f0ab3de-4130-4fc3-9ecf-8d372e7b2edc

**Deployments: Add, Edit works as expected. Tabs Rename pass too.**

https://github.com/user-attachments/assets/8e3a8ea7-5ef1-43e3-96d1-e52a9728b635

**Global: General mouse and keyboard navigation works as expected.**

https://github.com/user-attachments/assets/0b8d45ca-214c-4f14-9e34-2b1171d946e1

**Comment by izaac** [comment]
Rancher version: `v2.14-03c65adbbe9dedb99b1a938799b506ff390b3cf6-head`
Rancher UI version: `master 1f0b56c`
Browsers: Chrome `144` Firefox: `147` OS: `macOS Sequoia`/ `Linux 6.12` 

**Functional/Regressionl** tests:

- Cluster: `Create`, `Edit` configuration tabs behavior meets the expectation.
- Cluster: `Create`, `Edit` specific to node pools behavior regressions weren't observed.
- Deployments: `Add`, `Edit` works as expected. `Tabs Rename` pass too. 
- Global: General mouse and keyboard navigation works as expected.
